katas/content/superposition/greenberger_horne_zeilinger/Verification.qs

```diff
@@ -0,0 +1,20 @@
+namespace Kata.Verification {
+    open Microsoft.Quantum.Katas;
+    open Microsoft.Quantum.Arrays;
+
+    operation GHZ_State_Reference (qs : Qubit[]) : Unit is Adj + Ctl {
+        H(qs[0]);
+        
+        for q in Rest(qs) {
+            CNOT(qs[0], q);
+        }
+    }
+
+    @EntryPoint()
+    operation CheckSolution() : Bool {
+        CheckOperationsEquivalenceOnZeroStateWithFeedback(
```

Comment:
You'll want to loop over N here, from 1 to 5 or something like that, and check that the state prepared is correct for all lengths